### PR TITLE
Rocket

### PR DESCRIPTION
**Pull request by francksefu**
Redux: Fetch data and update Redux store -> done
Render UI:lists -> done
Write actions and reducers for booking rockets/dragons and joining missions -> done
Write actions and reducers for canceling rockets/dragons and leaving missions -> done
conditional components rendering -> done
My Profile section -> done

### CODE REVIEW

**Review by Ali-0111 — COMMENTED** [review]
Please follow this suggestions:

- [ ] Be prepare for  big changes 🤟 , I want to review your code. I am sure it will take more than 2- hours. 😄

**Review by Ali-0111 — CHANGES_REQUESTED** [review]
Required Changes:
please bring these changes;

File: src/components/RocketItem.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      <div className="flexible">
+        <div>
+          <img src={rocke.flickr_images[0]} alt="franck" />
+        </div>
+        <div>
+          <h2>{rocke.name}</h2>
+          <p>{rocke.description}</p>
+          { rocke.reserved
+            ? (<button type="button"> Cancel</button>) : (<button type="button" onClick={() => dispatch(reservation(rocke.id))}> Reserve Rocket</button>)}
+
```
Comment (Ali-0111) [inline src/components/RocketItem.js]: - [ ] to keep the code more readable, use  separated react components. Consider this pattern:
`{
condition ?  componentA : componentB
}`

File: src/reportWebVitals.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
-const reportWebVitals = onPerfEntry => {
+const reportWebVitals = (onPerfEntry) => {
   if (onPerfEntry && onPerfEntry instanceof Function) {
-    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
+    import('web-vitals').then(({
+      getCLS, getFID, getFCP, getLCP, getTTFB,
+    }) => {
       getCLS(onPerfEntry);
       getFID(onPerfEntry);
       getFCP(onPerfEntry);
```
Comment (Ali-0111) [inline src/reportWebVitals.js:8]: - [ ]  you can remove unnecessary codes from these lines

**Review by Ali-0111 — CHANGES_REQUESTED** [review]
Changes require, please see the comments below;

File: src/components/RocketItem.js (hunk 1 above)
Comment (francksefu) [reply src/components/RocketItem.js]: It's not necessary to add new components when the code to add is not complex ... keep it simple(Js good practice)

File: src/components/redux/rocket/rocketSlice.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    builder
+      .addCase(getRocket.fulfilled, (state, action) => {
+        state.rocket = action.payload;
+      })
+      .addCase(getRocket.pending, (state) => {
+        state.isLoading = true;
+      });
+  },
+});
+
```
[the remark this answers is not in the document]
Comment (francksefu) [reply src/components/redux/rocket/rocketSlice.js]: `return { ...state, rocket: action.payload, }; })` and `state.rocket = action.payload;` it's the same things, isn't violate the principle of immutability

**Pull request merged**